### PR TITLE
71: add 2 more cov structures (compound symmetry), add accept_singular arg

### PR DESCRIPTION
**Pull request by danielinteractive**
part of https://github.com/openpharma/mmrm/issues/71

To do:
- [x] Install `mmrm` from same branch name / PR
- [x] Install `tern.mmrm` from this branch / PR
- [x] Try out with the complicated data set @juliadedic1 :-) 
       (when I tried it out it did not work with unstructured, but with auto-regressive or compound symmetry, and results e.g. with lsmeans plot looked reasonable though sparse, due to missing data points at arm/visit combinations)

### CODE REVIEW

**Comment by juliadedic1** [comment]
@danielinteractive I'm not sure if I did something wrong but I keep getting this error message when I include, exclude or change the value from TRUE/FALSE for the `accept_singular` argument in `fit_mmrm`.

image

**Comment by danielinteractive** [comment]
@juliadedic1 now that the `mmrm` package is updated on its `main` branch please install from that before installing `tern.mmrm` from here

**Comment by dinakar29** [comment]
> @juliadedic1 now that the `mmrm` package is updated on its `main` branch please install from that before installing `tern.mmrm` from here

@juliadedic1 - do this:

```R
remotes::install_github("openpharma/mmrm", ref = "main", force = TRUE)
remotes::install_github("insightsengineering/tern.mmrm", ref = "71_non_full_rank", force = TRUE)
```

File: tests/testthat/test-fit_mmrm.R [hunk 1]
```diff
@@ -69,7 +73,8 @@ get_version <- function(version = c("A", "B")) {
   set.seed(123, kind = "Mersenne-Twister") # Because of `sample` below.
   mmrm_test_data %>%
     droplevels() %>%
-    { # nolint
+    {
```
Comment (dinakar29) [inline tests/testthat/test-fit_mmrm.R:76]: `styler` and `lintr` might conflict here. Best to `# nolint` this section

File: tests/testthat/test-fit_mmrm.R [hunk 2]
```diff
@@ -69,7 +73,8 @@ get_version <- function(version = c("A", "B")) {
...
+      # nolint
```
Comment (dinakar29) [inline tests/testthat/test-fit_mmrm.R:77]: That is, convert this to a "block style" no lint

**Review by juliadedic1 — APPROVED** [review]
Looks good! I got the exact same results as you @danielinteractive

**Comment by danielinteractive** [comment]
Thanks @dinakar29 for your help and @juliadedic1 for the review! Going to finish this later today

**Pull request merged**